code/modules/mob/living/blood_types.dm

```diff
@@ -0,0 +1,232 @@
+/datum/blood_type
+	/// Displayed name of the blood type.
+	var/name = "?"
+	/// A description of the blood type.
+	var/desc
+	/// Shown color of the blood type.
+	var/color = BLOOD_COLOR_RED
+	/// Additional lightness multiplier for the blood color, useful for when the default lightness from the greyscaling doesn't cut it and you want something more vibrant.
+	/// When set, color will be transformed into a matrix with coefficients multiplied by this value
+	var/lightness_mult = null
+	/// The cached color matrix for blood with a lightness_mult. We only need to calculate this once since blood types are singletons
+	var/list/blood_color_matrix
+	/// Blood types that are safe to use with people that have this blood type (for blood transfusions)
+	var/compatible_types = list()
+	/// What reagent is represented by this blood type?
+	var/datum/reagent/reagent_type = /datum/reagent/blood
+	/// What chem is used to restore this blood type (outside of itself, of course)?
+	var/datum/reagent/restoration_chem = /datum/reagent/iron
+	/// Whether or not this blood type should create blood trails, blood sprays, etc
+	var/no_bleed_overlays
+	/// If this is an abstract blood type, exclude it from being initialized
+	var/abstract
+
+/datum/blood_type/New()
+	. = ..()
+	compatible_types |= type_key()
+
+/datum/blood_type/Destroy(force)
+	if(!force)
+		stack_trace("qdel called on blood type singleton! (use FORCE if necessary)")
+		return QDEL_HINT_LETMELIVE
+
+	return ..()
+
+/**
+ * Key used to identify this blood type in compatible_types
+ *
+ * Allows for more complex or dynamically generated blood types
+ */
+/datum/blood_type/proc/type_key()
+	return type
+
+/// Returns blood color or color matrix
+/// Useful when you want to have a blood color with values out of normal hex bounds for that acidic look
+/// set dynamic to TRUE to redo the matrix each time (e.g. for clown blood dynamically shifting each time)
+/datum/blood_type/proc/get_color(dynamic = FALSE)
+	if(isnull(lightness_mult))
+		return color
+
+	if(!isnull(blood_color_matrix) && !dynamic)
+		return blood_color_matrix
+
+	blood_color_matrix = color_to_full_rgba_matrix(color)
+	for(var/i in 1 to min(length(blood_color_matrix), 16))
+		if (length(blood_color_matrix) == 12 && i > 9) // Don't modify constants
+			break
+		if (length(blood_color_matrix) >= 16 && i % 4 == 0) // Don't modify alpha either
+			continue
+		blood_color_matrix[i] *= lightness_mult
+
+	return blood_color_matrix
+
+// human blood type, for organizational purposes mainly
+/datum/blood_type/human
+	desc = "Blood cells suspended in plasma, the most abundant of which being the hemoglobin-containing red blood cells."
+	abstract = TRUE
+
+/datum/blood_type/human/a_minus
+	name = "A-"
+	compatible_types = list(/datum/blood_type/human/a_minus, /datum/blood_type/human/o_minus)
+	abstract = null
```

Comment:
TRUE vs null is a weird scheme. but also, why not just do abstract_type like station traits?